security/pkg/pki/ra/k8s_ra_test.go

```diff
@@ -249,12 +316,35 @@ func TestValidateCSR(t *testing.T) {
 	// Test Case 1
 	testSubjectIDs = []string{testCsrHostName, "Random-Host-Name"}
 	if !ValidateCSR(csrPEM, testSubjectIDs) {
-		t.Errorf("Test 1: CSR Validation failed")
+		t.Errorf("Test 1: CSR Validation failed. Expected success")
 	}
 
 	// Test Case 2
 	testSubjectIDs = []string{"Random-Host-Name"}
 	if ValidateCSR(csrPEM, testSubjectIDs) {
-		t.Errorf("Test 2: CSR Validation failed")
+		t.Errorf("Test 2: CSR Validation failed. CSR validation" +
+			" succeeded when expected failure due to mismatch in SANs")
```

Comment:
This is mostly what I am interested in - if we want to be opinionated with validation, we should have simple unit tests that express those opinions as comprehensively as possible. So the opinions cannot change without breaking at least one test.

Reply:
Totally agree. Will add simple unit test as well.

Reply:
@bleggett Added tests